test/widgets/content_test.dart

```diff
@@ -161,4 +161,33 @@ void main() {
       check(tester.takeException()).isA<AssertionError>();
     });
   });
+
+  group("CodeBlock", () {
+    Future<void> prepareContent(WidgetTester tester, String html) async {
+      await tester.pumpWidget(MaterialApp(home: BlockContentList(nodes: parseContent(html).nodes)));
+      await tester.pump();
+      await tester.pump();
+    }
+
+    testWidgets('with syntax highlighting', (WidgetTester tester) async {
+      // "```dart\nclass A {}\n```"
+      await prepareContent(tester,
+        '<div class="codehilite" data-code-language="Dart"><pre>'
+          '<span></span><code><span class="kd">class</span><span class="w"> </span>'
+          '<span class="nc">A</span><span class="w"> </span><span class="p">{}</span>'
+          '\n</code></pre></div>'
+      );
```

Comment:
nit: when an item in a comma-separated list (not only `List`, but any syntactic list of things) is followed by a newline, always use a comma:
```suggestion
          '\n</code></pre></div>',
      );
```

One has to do that anyway if it's any list item other than the last; so this trailing comma makes the syntax more symmetric, which is helpful for editing and for keeping diffs less noisy.

The other option here is to follow it directly with the close-paren instead of a newline:
```dart
        '\n</code></pre></div>');
```

Reply:
ack